WordPress/Classes/ViewRelated/Post/Categories/PostCategoriesViewController.swift

```
@@ -0,0 +1,281 @@
+        guard selectedCategories != nil else {
+            return
+        }
+
+        var didUpdateSelectedCategories = false
+        let updatedSelectedCategories = selectedCategories
+        self.selectedCategories = updatedSelectedCategories?.filter { category in
+            if let sortedCategories = blog.sortedCategories() as? [PostCategory], sortedCategories.contains(category), !category.isDeleted {
+                return true
+            } else {
+                didUpdateSelectedCategories = true
+                return false
+            }
+        }
+
+        // Notify the delegate of any changes for selectedCategories.
+        if didUpdateSelectedCategories {
+            delegate?.postCategoriesViewController?(self, didUpdateSelectedCategories: NSSet(array: selectedCategories!))
+        }
+
+        tableView.reloadData()
+    }
+
+    private func indentationLevelForCategory(parentID: NSNumber, categoryCollection: [NSNumber: PostCategory]) -> Int {
+        if parentID.intValue == 0 {
+            return 0
+        }
+
+        if let category = categoryCollection[parentID] {
+            return indentationLevelForCategory(parentID: category.parentID, categoryCollection: categoryCollection) + 1
+        } else {
+            return 0
+        }
+    }
+
+    //tableView
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        var result = categories.count
+        if selectionMode == .parent {
+            result = result + 1
+        }
+        return result
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.categoryCellIdentifier, for: indexPath) as? WPTableViewCell else {
+            return UITableViewCell()
```

It might be better to do `fatalError` here, rather than failing silently.

---

I know its more of the "Swift way" but I personally rather put the user experience first and not crash. How strongly do you feel about it?

---

Not too strongly – either way is an error and the categories view can't be used anyway 🤷‍♂️